src/codegen/compile.ml

```diff
@@ -5779,10 +6012,13 @@ module Stabilization = struct
         StableMem.write_word32 env ^^
 
         (* copy data to following stable memory *)
-        compile_const_64 4L ^^
-        extend64 get_dst ^^
-        extend64 get_len ^^
-        IC.system_call env "stable64_write"
+        StableMemoryStream.finalize_buffer
+          begin
+            compile_const_64 4L ^^
+            extend64 get_dst ^^
+            extend64 get_len ^^
+            IC.system_call env "stable64_write"
+          end
```

Comment:
```suggestion
          begin
            compile_const_64 4L ^^
            extend64 get_dst ^^
            extend64 get_len ^^
            IC.system_call env "stable64_write"
          end
```
Shouldn't this code be delete now?

Reply:
Sadly both `BumpStream` and `BlobStream` don't work for `Stabilisation` right now :-(
I'll investigate.

Reply:
Update: found the problem. f0d6437c0 fixes it. Now I can use either `BumpStream`, `BlobStream` or `StableMemoryStream` for `Externalization`.

Reply:
Cool.